src/index.ts

```diff
@@ -86,6 +89,94 @@ const {grpc} = new gax.GrpcClient();
  * @property {Constructor} [promise] Custom promise module to use instead of
  *     native Promises.
  */
+export interface Client {
+  //tslint: disable-next-line: no-any
+  [k: string]: any;
+  servicePath?: string;
+  sslCreds?: ChannelCredentials;
+  port?: number;
+}
+export interface ClientConfig {
+  [k: string]: string|{}|Function|Client|undefined;
+  apiEndpoint?: string;
+  projectId?: string;
+  appProfileId?: string;
+  email?: string;
+  keyFilename?: string;
+  credentials?: {
+    client_email: string;
+    private_key: string;
+  };
+  autoRetry?: boolean;
+  maxRetries?: number;
+  promise?: Constructor<{}>;
+}
+export interface Options extends ClientConfig {
+  BigtableClient?: Client;
+  BigtableInstanceAdminClient?: Client;
+  BigtableTableAdminClient?: Client;
+}
+
+export type CreateInstanceResponse = [IInstance, GaxOperation, LongrunningIOperation];
+export type GetInstanceResponse = [IInstance[], ListInstancesResponse];
+
+export type ICluster = btAdminClient.bigtable.admin.v2.ICluster;
+export type LongrunningIOperation = btAdminClient.longrunning.IOperation;
+export type ListInstancesResponse = btAdminClient.bigtable.admin.v2.IListInstancesResponse;
+export type InstanceState = btAdminClient.bigtable.admin.v2.Instance.State;
+export type InstanceType = btAdminClient.bigtable.admin.v2.Instance.Type;
+
+export interface CreateInstanceCallback {
+  (instance?: Instance, operation?: GaxOperation, apiResponse?: LongrunningIOperation): void;
+}
+export interface GetInstanceCallback {
+  (err?: Error|null, instances?: Instance[], apiResponse?: ListInstancesResponse): void;
+}
+export interface GetProjectIdCallback {
+  (err: Error|null, projectId?: string|null): void;
+}
+
+export interface CreateInstanceCluster {
+  id: string;
+  location: string;
+  nodes: number;
+  storage: string;
+}
+export interface CreateInstanceClusters {
+  [k: string]: {
+    location: string;
+    serveNodes: number;
+    defaultStorageType: ICluster;
+  };
+}
+export interface CreateInstanceOptions {
+  clusters?: CreateInstanceCluster[];
+  displayName?: string;
+  labels?: string|{[k: string]: string};
+  type?: string;
+  gaxOptions?: CallOptions;
+}
+export interface CreateInstanceRequest {
+  parent?: string|null;
+  instanceId?: string|null;
+  clusters?: ({[k: string]: ICluster}|null);
+  instance?: IInstance|null;
+}
+//tslint:disable-next-line interface-name
+export interface IInstance {
+  name?: string|null;
+  displayName?: string|null;
+  state?: InstanceState|null;
+  type?: InstanceType|null;
+  labels?: {[k: string]: string}|null|string;
+}
+export interface RequestConfig {
```

Comment:
@callmehiphop I feel like this interface has to be defined somewhere else already

Reply:
There is the [`ProtoOpts`](https://github.com/googleapis/nodejs-common-grpc/blob/master/src/service.ts#L38) interface in `grpc-common`, but we'll need to extend it (I think). For gax/handwritten clients, the request method is not inherited from anything, so that's probably why it isn't defined in any core libs.

Reply:
I hear ya - I extended `ProtoOpts` in OPR 1.2 but had some trouble with it so I added `ProtoOpts` properties into `RequestConfig`. I can slim it down if needed 👍